packages/domain-picker/src/components/index.tsx

```diff
@@ -160,9 +160,14 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		retryRequest: retryDomainSuggestionRequest,
 	} = useDomainSuggestions( domainSearch.trim(), quantityExpanded, domainCategory, locale ) || {};
 
-	const domainSuggestions = allDomainSuggestions?.slice(
-		existingSubdomain ? 1 : 0,
-		isExpanded ? quantityExpanded : quantity
+	// don't list the already existing free domain as part of the suggestions
+	const domainSuggestions = allDomainSuggestions
+		?.filter( ( suggestion ) => suggestion.domain_name !== existingSubdomain?.domain_name )
+		.slice( 0, isExpanded ? quantityExpanded : quantity );
```

Comment:
@StefanNieuwenhuis @alshakero 
Was this change needed in the context of this PR or it was just cleanup/refactoring?
I'm asking because it has introduced a regression in Focused Launch. See p1617903103117300-slack-C013QHLF28Y

Here is a fix (reverting this change): https://github.com/Automattic/wp-calypso/pull/51806

Reply:
I did a little debugging for `domainSuggestions`:
1. filter out all suggestions with a `domain_name` equal to the existing subdomain.
2. slice the filtered list to the predefined length.

While testing (stefannew.wordpress.com), my domain suggestions are:
* `stefannew319442470.wordpress.com`.
* `stefan-new.com`.
* `stefannew.life`.
* `stefannew.blog`.

The existing subdomain is: `stefannew.wordpress.com` so the free subdomain suggestion isn't filtered from the list resulting in the regression!

The previous implementation didn't care about `domain_name` and **always** cuts out the free subdomain when available. 

### Conclusion
* **Current implementation**: More specific but causes regressions when the names of the free subdomains don't match.
* **Previous implementation**:  Always cuts out the first suggestion when a free subdomain is available. 

It seems to me that both approaches try to achieve the same so I think it's safe to assume that this change was made to improve the domain suggestions filtering logic.

Since the latter doesn't cause regressions (for now) and provides a solution for p1617903103117300-slack-C013QHLF28Y, I think it's okay to apply the revert (#51806).

Reply:
Yes it was needed in this PRs context. Because this PR broke two assumptions we used to make:

1. The free domain is always the first suggestion, so we just shifted the array to remove it. This assumption becomes false when we added the unavailable suggestion at the top when users search for an FQDN.
2. That the second item of the array is the recommended domain (the first paid domain).

This change was an attempt at:

1. Find the free domain no matter where it is.
2. Find the first occuring paid domain in the suggestion.

Reply:
Thanks for elaborating!